shell/platform/common/text_editing_delta.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_TEXT_EDITING_DELTA_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_TEXT_EDITING_DELTA_H_
+#define _SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING
+
+#include <codecvt>
+#include <string>
+
+#include "flutter/shell/platform/common/text_range.h"
+
+namespace flutter {
+
+struct TextEditingDelta {
+ public:
+  TextEditingDelta(std::u16string textBeforeChange,
+                   TextRange range,
+                   std::u16string text);
+  virtual ~TextEditingDelta();
+
+  std::string oldText() const { return utf16ToUtf8(oldText_); }
+  std::string deltaText() const { return utf16ToUtf8(deltaText_); }
+  int deltaStart() const { return deltaStart_; }
+  int deltaEnd() const { return deltaEnd_; }
```

Comment:
Here and below, methods and functions should be named in `UpperCamelCase()`. 

Simple property getters/setters can be named `foo_bar()` and `set_foo_bar()`.

Ref: https://google.github.io/styleguide/cppguide.html#Function_Names